classes/Context.php

```diff
@@ -364,25 +364,30 @@ public function updateCustomer(Customer $customer)
             $this->cart = new Cart($idCart);
             $this->cart->secure_key = $customer->secure_key;
         } else {
-            $idCarrier = (int) $this->cart->id_carrier;
-            $this->cart->secure_key = $customer->secure_key;
-            $this->cart->id_carrier = 0;
-            $this->cart->setDeliveryOption(null);
-            $this->cart->updateAddressId($this->cart->id_address_delivery, (int) Address::getFirstCustomerAddressId((int) ($customer->id)));
-            $this->cart->id_address_delivery = (int) Address::getFirstCustomerAddressId((int) ($customer->id));
-            $this->cart->id_address_invoice = (int) Address::getFirstCustomerAddressId((int) ($customer->id));
+            if (Validate::isLoadedObject($this->cart)) {
+                $idCarrier = (int) $this->cart->id_carrier;
+                $this->cart->secure_key = $customer->secure_key;
+                $this->cart->id_carrier = 0;
+                if ($idCarrier) {
```

Comment:
Best practice is using strict boolean condition
```suggestion
                if (!empty($idCarrier)) {
```